docs/proposals/improved-schema-for-services.md

```diff
@@ -0,0 +1,562 @@
+                "apiVersion": {
+                  "type": "string"
+                },
+                "fieldPath": {
+                  "type": "string"
+                }
+              }
+            },
+            "resourceFieldRef": {
+              "type": "object",
+              "required": [
+                "resource"
+              ],
+              "properties": {
+                "containerName": {
+                  "type": "string"
+                },
+                "divisor": {
+                  "type": "string"
+                },
+                "resource": {
+                  "type": "string"
+                }
+              }
+            },
+            "secretKeyRef": {
+              "type": "object",
+              "required": [
+                "key"
+              ],
+              "properties": {
+                "key": {
+                  "type": "string"
+                },
+                "name": {
+                  "type": "string"
+                },
+                "optional": {
+                  "type": "boolean"
+                }
+              }
+            }
+          }
+        }
+      }
+    }
+  },
+  "detectBindingResources": {
+    "type": "boolean"
+  },
+  "envVarPrefix": {
+    "type": "string"
+  },
+  "mountPathPrefix": {
+    "type": "string"
+  },
+  "services": {
+    "type": "array",
+    "items": {
+      "type": "object",
+      "required": [
+        "group",
+        "kind",
+        "version"
+      ],
+      "properties": {
+        "envVarPrefix": {
+          "type": "string"
+        },
+        "group": {
+          "type": "string"
+        },
+        "id": {
+          "type": "string"
+        },
+        "kind": {
+          "type": "string"
+        },
+        "name": {
+          "type": "string"
+        },
+        "namespace": {
+          "type": "string"
+        },
+        "version": {
+          "type": "string"
+        }
+      }
+    }
+  }
+}
+
+```
+
+Note - removed `description` fields to make the sample concise. 
+</details>
+
+
+- There are 5 top level fields in the metadata -
+    - `application`
+    - `customEnvVar`
+    - `detectBindingResources`
+    - `envVarPrefix`
+    - `mountPathPrefix`
+    - `services`
+
+### Conversion of openAPIV3Schema to json schema
+
+We after discussions with adapters team have decided to convert the openAPIV3Schema to json schema so that it would be easy to consume and hence we would be working on the converted json schema version to derive information from it.
+
+TODO: find a good enough library to convert openAPIV3Schema to json schema reliably 
+
+### cobra flags to golang map conversion
+
+We need to finalise a mechanism/syntax so that a user can set any property for a CR using the command line, for e.g. if we considered `jsonpath` something like this `odo service create servicebinding.coreos.io/Servicebinding -services[0].envVarPrefix "SVC" -services[0].namespace "openshift"`. 
```

Comment:
I'm afraid that this is not possible in Cobra. It will have to be converted into something like `-p "services[0].envVarPrefix"="SVC"`.

`servicebinding.coreos.io/Servicebinding` is not unique identification of a CRD. The cluster can have multiple CRDs with kind `Servicebinding` and group `servicebinding.coreos.io` in different versions and each version can have a different structure. The unique identification should include full `apiVersion` (group/version) and `kind`, for example `operators.coreos.com/v1alpha1/ServiceBinding`.

Because this is a  lot of typing odo should be smart. If there is only one CRD version `operators.coreos.com/ServiceBinding` should work. If there are multiple versions it needs to error out and ask the user to provide a version eg. `operators.coreos.com/v1alpha1/ServiceBinding`

Reply:
added the actual approach